exercicios/para-casa/index.js

```diff
@@ -0,0 +1,92 @@
+const express = require('express')
+const app = express()
+const listaUsuarios = require("./model/usuarios.json")
+app.use(express.json())
+const port = 3000
+
+app.put("/usuarios/:id", (req, res) => {
+    const Idcadastros = req.params.id
+    const usuarioAtualizado = req.body
+
+    const temUsuario = listaUsuarios.find(usuarios => usuarios.id == Idcadastros)
+
+    if (temUsuario) {
+        return res.status(200).json(usuarioAtualizado)
+    }
+
+    listaUsuarios.push(usuarioAtualizado)
+
+    return res.status(201).json(usuarioAtualizado)
+
+})
+
+
+app.patch("/usuarios/:id", (req, res) => {
+    const Idcadastros = req.params.id
+    const camposAtualizados = req.body
+
+    const existeEnderenco = listaUsuarios.find(usuarios => usuarios.id == Idcadastros)
+
+    if (existeEnderenco) {
+        const enderecoAtualizado = {
+            ...existeEnderenco,
+            ...camposAtualizados
+        }
+        listaUsuarios.map((usuarios, index)=>{
+            if(usuarios.id == Idcadastros){
+                return listaUsuarios[index] = enderecoAtualizado
+            }
+        })
+
+        return res.status(200).json(enderecoAtualizado)
+    }
+
+    return res.status(404).json({
+        messagem: "Usuario não Encontrado, por favor verifique."
+    })
+})
+
+/*  MEU PRIMEIRO CODIGO DELETE VERSÃO 1.0
```

Comment:
Tudo bem deixar anotações. Mas lembre-se que quando for para produção, apenas o seu código importa. Ok?